scanner/src/test/java/ca/on/oicr/gsi/runscanner/scanner/processor/AbstractProcessorTest.java

```diff
@@ -29,7 +31,24 @@ protected final void checkDirectory(String root)
           null); // We delete this because it is going to change during updates to dependencies.
       result.setSequencerFolderPath(
           null); // We delete this because it is going to be different in each environment.
-      result.setMetrics(null); // We delete these because changes in metrics are non-critical.
+
+      // For Illumina processors only
+      if (clazz.equals(IlluminaNotificationDto.class)) {
+        // Load the metrics into a Jackson JsonNode to test for equality
+        ObjectMapper MapperTest = new ObjectMapper();
+        JsonNode jsonNodeResult = MapperTest.readTree(result.getMetrics());
+        JsonNode jsonNodeReference = MapperTest.readTree(reference.getMetrics());
+        if (!jsonNodeReference.equals(jsonNodeResult)) {
+          assertEquals(reference, result);
+        }
```

Comment:
You're already checking `reference` vs. `result` below. You can just `assertEquals(jsonNodeReference, jsonNodeResult)` here